hledger-lib/Hledger/Read.hs

```diff
@@ -126,8 +127,11 @@ defaultJournalPath = do
 --
 readJournalFiles :: Maybe StorageFormat -> Maybe FilePath -> Bool -> [PrefixedFilePath] -> IO (Either String Journal)
 readJournalFiles mformat mrulesfile assrt prefixedfiles = do
-  (either Left (Right . mconcat) . sequence)
+  (right mconcat1 . sequence)
     <$> mapM (readJournalFile mformat mrulesfile assrt) prefixedfiles
+  where mconcat1 :: Monoid t => [t] -> t
+        mconcat1 [] = mempty
+        mconcat1 x = foldr1 mappend x
```

Comment:
I often hear foldr is the wrong fold to use.. is it the right one here, do you think ?